source/validate_id.cpp

```diff
@@ -2387,32 +2415,38 @@ namespace libspirv {
 /// checked during the initial binary parse in the IdPass below
 spv_result_t CheckIdDefinitionDominateUse(const ValidationState_t& _) {
   for (const auto& definition : _.all_definitions()) {
-    // Check only those blocks defined in a function
-    if (const Function* func = definition.second.defining_function()) {
-      if (const BasicBlock* block = definition.second.defining_block()) {
+    // Check only those definitions defined in a function
+    if (const Function* func = definition.second->function()) {
+      if (const BasicBlock* block = definition.second->block()) {
+        if (!block->reachable()) continue;
         // If the Id is defined within a block then make sure all references to
         // that Id appear in a blocks that are dominated by the defining block
-        for (auto use : definition.second.uses()) {
-          if (!use->reachable()) continue;
-          if (use->dom_end() == find(use->dom_begin(), use->dom_end(), block)) {
-            return _.diag(SPV_ERROR_INVALID_ID)
-                   << "ID " << _.getIdName(definition.first)
-                   << " defined in block " << _.getIdName(block->id())
-                   << " does not dominate its use in block "
-                   << _.getIdName(use->id());
+        for (auto& use_index_pair : definition.second->uses()) {
+          const Instruction* use = use_index_pair.first;
+          if (const BasicBlock* use_block = use->block()) {
+            if (!use_block->reachable()) continue;
+            if (use->opcode() != SpvOpPhi &&
```

Comment:
The old OpPhi special case still checked the def-use of the result type.   :-(
But I guess I didn't have a test case for that?